src/darwin/Framework/CHIP/MTRMetricKeys.h

```diff
@@ -22,6 +22,16 @@ namespace chip {
 namespace Tracing {
 namespace DarwinFramework {
 
+// When metrics were originally added, they were used for logging events during the commissioning of devices. The data was sent at
+// the end of commissioning (in OnCommissioningComplete). This means that if we want to add metrics that are not related to
+// commissioning, that need to be sent outside of the commissioning of the device, we need a way to separate out metrics by a
+// category. The CHIP metrics infrastructure is fairly simple, with just a key and a numeric value, so to add categories we've
+// opted to use a special prefix (dwnfw__CATEGORY__) for the keys that are in a category. Any keys that do not contain this special
+// prefix are assumed to be used for commissioning events (because we don't want to rename the existing keys).
+
+#define NOT_COMMISSIONING_METRICS_PREFIX "dwnfw__"
```

Comment:
Why not rename to METRICS_KEY_PREFIX and METRICS_KEY and just document this is the new way to define new metric keys. There is a oddness for (MTRMetrics *)metricSnapshot:(BOOL)resetCollection, but we just document that there as to why we are dropping keys with METRICS_KEY_PREFIX